perun-base/src/main/java/cz/metacentrum/perun/registrar/model/ApplicationFormItem.java

```diff
@@ -251,7 +251,15 @@ public void setUpdatable(boolean updatable) {
 		 * Special type for specifying if will be allowed to register to group(s) through VO application form. This type
 		 * is represented by standard HTML checkbox.
 		 */
-		EMBEDDED_GROUP_APPLICATION
+		EMBEDDED_GROUP_APPLICATION,
+		/**
+		 * Widgets for array like values.
+		 */
+		LIST_INPUT_BOX,
+		/**
+		 * Widgets for map like values (key - value pairs).
+		 */
+		MAP_INPUT_BOX
```

Comment:
Maybe I would specify how values of such items should look like - separated by spaces and colons. And should special characters (`,` or `:`) be escaped so we could actually process such values - e.g., an array with 3 items like this: `item,item\,withComma,anotherItem`?

Reply:
Sorry, I meant separated by **commas** and colons.

Reply:
I have added a better specification based on the `stringToAttributeValue` method in `BaseUtils`.